diff --git a/src/os_unix.c b/src/os_unix.c
index fdae5d67f..46064d172 100644
--- a/src/os_unix.c
+++ b/src/os_unix.c
@@ -6070,9 +6070,6 @@ mch_job_start(char **argv, job_T *job, jobopt_T *options, int is_terminal)
 		// In the GUI 'term' is not what we want, use $TERM.
 		term = getenv("TERM");
 #endif
-	    // Use 'term' or $TERM if it starts with "xterm", otherwise fall
-	    // back to "xterm" or "xterm-color".
-	    if (term == NULL || *term == NUL || STRNCMP(term, "xterm", 5) != 0)
 	    {
 		if (t_colors >= 256)
 		    // TODO: should we check this name is supported?
